zowex/src/main.rs

```diff
@@ -129,59 +132,40 @@ fn main() -> std::io::Result<()> {
     }
 }
 
-/**
- * Convert a vector of command line arguments into a single string of arguments.
- * @param cmd_line_args
- *      The user-supplied command line arguments to the zowe command.
- *      Each argument is in its own vector element.
- * @returns
- *      A String containing all of the command line arguments.
- */
-fn arg_vec_to_string(arg_vec: Vec<String>) -> String {
-    let mut arg_string = String::new();
-    let mut arg_count = 1;
-    for next_arg in arg_vec.iter() {
-        if arg_count > 1 {
-            arg_string.push(' ');
-        }
-
-        /* An argument that contains a space, or is an empty string, must be
-         * enclosed in double quotes when it is placed into a single argument
-         * string.
-         */
-        if next_arg.contains(' ') || next_arg.len() == 0 {
-            arg_string.push('"');
-            arg_string.push_str(next_arg);
-            arg_string.push('"');
-        } else {
-            arg_string.push_str(next_arg);
-        }
-
-        arg_count = arg_count + 1;
-    }
-
-    return arg_string;
+fn get_zowe_env() -> HashMap<String, String> {
+    env::vars().filter(|&(ref k, _)|
+        k.starts_with("ZOWE_")
+    ).collect()
 }
 
-fn run_daemon_command(mut args: String) -> std::io::Result<()> {
-    let path = env::current_dir()?;
-    args.insert(0, '\r');
-    args.insert_str(0, path.to_str().unwrap());
-    let mut _resp = args.as_bytes(); // as utf8 bytes
-
-    if _resp.is_empty() {
-        _resp = b" ";
+fn run_daemon_command(args: &mut Vec<String>) -> std::io::Result<()> {
+    let cwd = env::current_dir()?;
+    let mut stdin = String::new();
+    if !atty::is(Stream::Stdin) {
+        io::stdin().read_to_string(&mut stdin)?;
+    }
+    let response: DaemonResponse = DaemonResponse {
+        argv: Some(args.to_vec()),
+        cwd: Some(cwd.into_os_string().into_string().unwrap()),
+        env: Some(get_zowe_env()),
+        stdinLength: Some(stdin.len() as i32),
+        stdin: None,
+    };
+    let mut _resp = serde_json::to_string(&response)?;
+    if response.stdinLength.unwrap() > 0 {
+        _resp.push('\n');
```

Comment:
Curious if this may need changed based on the previous comment.
Please disregard this comment if this is not where the `}\n` lookup is coming from.